exporter/otlpexporter/otlp.go

```diff
@@ -110,7 +116,12 @@ func (e *baseExporter) pushMetrics(ctx context.Context, md pmetric.Metrics) erro
 	}
 	partialSuccess := resp.PartialSuccess()
 	if !(partialSuccess.ErrorMessage() == "" && partialSuccess.RejectedDataPoints() == 0) {
-		return consumererror.NewPermanent(fmt.Errorf("OTLP partial success: \"%s\" (%d rejected)", resp.PartialSuccess().ErrorMessage(), resp.PartialSuccess().RejectedDataPoints()))
+		// TODO: These should be counted as "rejected", similar to dropped items.
+		// https://github.com/open-telemetry/opentelemetry-collector/issues/9243
+		e.settings.Logger.Warn("partial success",
+			zap.String("message", resp.PartialSuccess().ErrorMessage()),
+			zap.Int64("num_rejected", resp.PartialSuccess().RejectedDataPoints()),
```

Comment:
I think we should be consistent with other log messages where we use `dropped_data_points`, `dropped_spans` and `dropped_records` as the key. We can use the `rejected` term, but so far, it's been used in logs for retryable errors propagated to the client, and `dropped` is used for permanent errors.

Reply:
I'd like us to find consistent terminology. From my review in https://github.com/open-telemetry/oteps/pull/238, I find that "dropped" most consistently refers to data that was never send to the downstream service. The term "rejected" is already standardized in the PartialSuccess message.

I would not like us to continue using "dropped" to refer to data that was sent and failed. That data "failed" after being sent to a consumer, so it wasn't dropped. There's even a chance the consumer successfully received that data, but that a timeout occurred, leaving the situation unclear, with success on the server but timeout at the client. I would like "dropped" to refer to data that was not transmitted, because it is helpful in constructing pipeline equations for monitoring. 

The logic I want to enforce is that whatever is received is either sent or dropped. If the data was sent, it can fail or not, but it wasn't dropped until after it was sent (and the receiver could have processed it anyway).

Reply:
Since the error before this change was using rejected, and the current change only prefixes this w/ `num_`, could we move the discussion on standardizing terminology to a separate issue? @dmitryax

Reply:
From https://github.com/open-telemetry/oteps/pull/238, it's still not very clear how to distinguish permanent failures from retriable. We use `dropped` and `rejected` in the collector accordingly, even though it wasn't applied consistently.

In this particular case, the `rejected` term makes sense from the standpoint of a server sending this response, in my opinion. But in the collector, we drop that portion of the data.

Otherwise, the PR looks good to me. I'm not going to block it just because of the zap log key.

Reply:
I've updated this PR to be consistent as requested by @dmitryax. However, I'll note that exporterhelper is not fully consistent with itself. When data drops after retry attempts, it is `dropped_items` independent of the signal type, whereas the other type of failure has three specific keys.

I'll keep working to clarify https://github.com/open-telemetry/oteps/pull/238, thanks for the input. I don't agree with the categorization of "rejected" and "dropped", but we can discuss it there. Is there any more documentation on this choice of words? Even as I read the code for exporterhelper, I can't quite make sense of it. I see one log message from the baseExporter,
```
	if err != nil {
		be.set.Logger.Error("Exporting failed. Rejecting data."+be.exportFailureMessage,
			zap.Error(err), zap.Int("rejected_items", req.ItemsCount()))
	}
```
one for the queueSender:
```
		consumeErrHandler := func(err error, req Request) {
			o.set.Logger.Error("Exporting failed. Dropping data."+o.exportFailureMessage,
				zap.Error(err), zap.Int("dropped_items", req.ItemsCount()))
		}
```
and one for the converter
```
			set.Logger.Error("Failed to convert logs. Dropping data.",
				zap.Int("dropped_log_records", ld.LogRecordCount()),
				zap.Error(err))
```
What do you call it when a sampler drops data? How is that the same / not the same as these cases? I see the sampler-dropping case as similar in outcome to the converter failure -- you couldn't convert the data so you never sent it -- the coinflip was negative so you never sent it. I see the sampler-dropping case as dissimilar to the queueSender and baseExporter outcome, because the data was sent.

My position is that metrics should only count after the final outcome is known, so including attempts to retry, meaning that retries only delay the outcome.  I only count outcomes once per point/span/record. Therefore, retries are not visible in pipeline metrics -- they will be visible in transport-specific tracing and metrics instrumentation. So, e.g., you'll see two RPCs if there is one retry, you'll see twice as many bytes sent if you have to send a request twice due to retries, and so on.

Reply:
"Exporting failed. Rejecting data" is logged with the error returned to the caller. So it can be retried potentially.

"Exporting failed. Dropping data." in the queueSender is logged on the data consumed from the queue asynchronously, so there is no way to retry them.

"Failed to convert logs. Dropping data." is returned with the permanent error, so they should not be retried.